test/jdk/sun/security/util/AlgorithmConstraints/DisabledAlgorithmPermits.java

```diff
@@ -0,0 +1,180 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import sun.security.util.DisabledAlgorithmConstraints;
+
+import java.security.CryptoPrimitive;
+import java.security.KeyPairGenerator;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/*
+ * @test
+ * @bug 8346129
+ * @modules java.base/sun.security.util
+ * @summary Check DisabledAlgorithmConstraints using EdDSA & XDH against
+ * permit(Set<CryptoPrimitive>, String, AP) and
+ * permit(Set<CryptoPrimitive>, Key).  Results will differ based
+ * on method used.  The first method only can compare the String with the
+ * algorithms.  The second can check the algorithm and NamedParameterSpec
+ * while results in more 'false' cases.
+ *
+ * @run main/othervm DisabledAlgorithmPermits Ed25519
+ * @run main/othervm DisabledAlgorithmPermits Ed448
+ * @run main/othervm DisabledAlgorithmPermits EdDSA
+ * @run main/othervm DisabledAlgorithmPermits X25519
+ * @run main/othervm DisabledAlgorithmPermits X448
+ * @run main/othervm DisabledAlgorithmPermits XDH
+ */
+
+public class DisabledAlgorithmPermits {
+    public static void main(String[] args) throws Exception {
+        String algorithm = args[0];
+        Security.setProperty("x", algorithm);
+
+        // Expected table are the expected results from the test
+        List<TestCase> expected = switch (algorithm) {
+            case "Ed25519" ->
+                Arrays.asList(
+                    new TestCase("EdDSA", true),
+                    new TestCase("Ed25519", false),
+                    new TestCase("Ed448", true),
+                    new TestCase("X448", true),
+                    new TestCase(1,"EdDSA", false),
+                    new TestCase(1,"Ed25519", false),
+                    new TestCase(1,"Ed448", true),
+                    new TestCase(1,"X448", true));
+            case "Ed448" ->
+                Arrays.asList(
+                    new TestCase("EdDSA", true),
+                    new TestCase("Ed25519", true),
+                    new TestCase("Ed448", false),
+                    new TestCase("X448", true),
+                    new TestCase(1,"EdDSA", true),
+                    new TestCase(1,"Ed25519", true),
+                    new TestCase(1,"Ed448", false),
+                    new TestCase(1,"X448", true));
+            case "EdDSA" ->
+                Arrays.asList(
+                    new TestCase("EdDSA", false),
+                    new TestCase("Ed25519", true),
```

Comment:
Why should the above pass? If you disable `EdDSA` and you are still allowed `Signature.getInstance("EdDSA")`? If this is because it will reject whatever EdDSA key later? Why both check `CryptoPrimitive.SIGNATURE` at all?

Reply:
I'm confused by this comment.  With removing the hardcoded aliases in AbstractAlgorithmConstraints, which is what I thought you had suggested, EdDSA and Ed25519 are now separate as the check is effectively a string compare check against the disabledAlgorithm list

The second half of that case statement has a key that can check against both EdDSA and the NPS.

With respect to `CryptoPrimitive.SIGNATURE`, it just a value used in the test, it can't be null.

Reply:
Yes, I remember that.

I understand there will be multiple checks in TLS and CertPath. Do we have existing tests on that level?